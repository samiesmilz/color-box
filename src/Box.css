.Box {
  border: 1px solid rgb(71, 96, 87);
  border-radius: 5px;
  margin: 10px;
  padding: 10px;
}
.Box-color {
  border: 1px solid black;
  border-radius: 5px;
  margin: 10px;
}
.Box-label {
  font-size: smaller;
  padding: 10px;
  text-align: left;
}
