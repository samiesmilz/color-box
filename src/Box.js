import React from "react";
import "./Box.css";

const Box = ({ width, height, color, id, handleRemove }) => {
  const remove = () => {
    handleRemove(id);
  };
  return (
    <div className="Box">
      <p className="Box-label">
        This is a box with width of {width}px and height of {height}px with
        color - {color}.
      </p>
      <div
        className="Box-color"
        style={{
          backgroundColor: color,
          width: `${width}px`,
          height: `${height}px`,
        }}
      >
        <button onClick={remove}> ❇️X </button>
      </div>
    </div>
  );
};

export default Box;
