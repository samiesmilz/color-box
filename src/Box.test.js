import { render, screen } from "@testing-library/react";
import Box from "./Box";

it("renders withou errors", () => {
  render(<Box />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});
