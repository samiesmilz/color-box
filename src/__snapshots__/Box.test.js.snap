// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <p
      class="Box-label"
    >
      This is a box with width of px and height of px with color - .
    </p>
    <div
      class="Box-color"
    >
      <button>
         ❇️X 
      </button>
    </div>
  </div>
</DocumentFragment>
`;
